chain/network/src/routing.rs

```diff
@@ -546,29 +560,47 @@ impl RoutingTable {
     }
 
     pub fn add_ping(&mut self, ping: Ping) {
-        if self.ping_info.is_none() {
-            self.ping_info = Some(HashMap::new());
-        }
+        self.ping_info.cache_set(ping.nonce as usize, ping);
+    }
+
+    /// Return time of the round trip of ping + pong
+    pub fn add_pong(&mut self, pong: Pong) -> Option<f64> {
+        let mut res = None;
 
-        if let Some(ping_info) = self.ping_info.as_mut() {
-            ping_info.entry(ping.nonce as usize).or_insert(ping);
+        if let Some(nonces) = self.waiting_pong.cache_get_mut(&pong.source) {
+            res = nonces
+                .cache_remove(&(pong.nonce as usize))
+                .and_then(|sent| Some(Instant::now().duration_since(sent).as_secs_f64() * 1000f64));
```

Comment:
why not use `as_millis` and also why do you need floating point numbers here?

Reply:
Because it returns `usize` instead of `f64`.

Reply:
Good point about floating points, using it because ultimately I'm finding mean and weighted mean, so I avoid using integers, but at this point `usize` will do the work.

Reply:
After further looking into `as_millis` vs `as_secs_f64` the latter has more precision than the former, so leaving it how it is.

Reply:
@mfornet there are also `as_micros` and `as_nanos` if you need precision. Though, all of those return u128, which might have their own implications.

Reply:
Yeh, already saw those, ultimately I'm converting to f64 so, I think it is fine having this value as f64 from the beginning.